compiler-plugin/build.gradle.kts

```diff
@@ -19,6 +19,7 @@ repositories {
 
 dependencies {
     compileOnly(libs.kotlin.compiler.embeddable)
+    compileOnly(libs.kotlin.reflect)
```

Comment:
Do we need this dependency?

Reply:
Also wait, why compileOnly?

Reply:
@FreshMag If we do not use the kotlin reflection anymore can we remove it, right?

Reply:
> @FreshMag If we do not use the kotlin reflection anymore can we remove it, right?

The implementation that uses the Kotlin reflection is still in there. I've marked it as `@Deprecated`. I extracted also the function that uses Java into a separate function, that's the one used right now. 
If you prefer I'll directly remove the Kotlin one and remove the dependency of Kotlin reflect as well.

> Also wait, why compileOnly?

That's probably my mistake. I'll change it.